File: series2/series2_handout/ssprk3_timestep/make_stability.py
from numpy import *
import os
import matplotlib
import matplotlib.pyplot as plt

for dt in [2.5, 2, 1.5, 1, 0.5]:
    os.system("./ssprk3 %f" % dt)
    t = loadtxt('time.txt')
    u = loadtxt('solution.txt')
    plt.plot(t, u, '-*', label='$u(t)$')
    plt.xlabel('$t$')
    plt.ylabel('$u(t)$')
    plt.legend()
    plt.savefig('solution_%d.png' % int(10*dt))
    plt.close('all')

File: series2/series2_code/series2_code/ssprk3_timestep/make_stability.m
for dt = [2.5, 2, 1.5, 1, 0.5]
    f = figure('visible','off');
    cmd = ['./ssprk3 ', num2str(dt)];
    status = system(cmd);
    if status ~= 0
        disp('You should be running this from the build directory. Are you?')
        return
    end
    t = load('time.txt');
    u = load('solution.txt');
    plot(t, u, '-*', 'DisplayName','u(t)');
    hold on;
    xlabel('t')
    ylabel('u(t)')
    legend;
    outfile = ['solution_', num2str( floor(dt*10)), '.png'];
    saveas(gcf,outfile);
    close all;
end
